src/main/java/eu/stamp_project/testrunner/EntryPoint.java

```diff
@@ -394,26 +357,30 @@ private static void runGivenCommandLine(String commandLine) throws TimeoutExcept
             );
             workingDirectory = null;
         }
-        ExecutorService executor = Executors.newSingleThreadExecutor();
-        Process process = null;
-        try {
-            process = Runtime.getRuntime().exec(commandLine, null, workingDirectory);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-        final Future<?> submit = executor.submit(new RunnableProcess(process, commandLine));
-        try {
-            submit.get(timeoutInMs, TimeUnit.MILLISECONDS);
-        } catch (TimeoutException e) {
-            throw e;
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        } finally {
+    	Process process = null;
+		try {
+			ProcessBuilder pb = new ProcessBuilder().command(Arrays.asList(commandLine.split(" ")));
+			if (workingDirectory!=null) {
+				pb.directory(workingDirectory);
+			}
+
+			if (EntryPoint.verbose) {
+				process = pb.inheritIO().start();
```

Comment:
If I get it, we lose the feature to give a custom PrintStream in order to collect the output and the error? See [here](https://github.com/STAMP-project/testrunner/blob/master/src/main/java/eu/stamp_project/testrunner/EntryPoint.java#L98) and [here](https://github.com/STAMP-project/testrunner/blob/master/src/main/java/eu/stamp_project/testrunner/EntryPoint.java#L104).

Reply:
No, this is managed in proposed code, see lines:
```
if (EntryPoint.verbose) {
    process = pb.inheritIO().start();
}
```
The pb.inheritIO() redirects this process IO stream to the caller IO stream. I've tested and indeed it works much better than before, since it also redirect logs printed in tests using Log4j, while before it only printed logs sent to System.out

Reply:
> The pb.inheritIO() redirects this process IO stream to the caller IO stream.

Which means we cannot capture the IO in a custom PrintStream. What I mean is, if `EntryPoint.verbose` is true, we enable the redirection of the IO. By default, it redirects the IO on the caller's IO, right? I understand that this is handled by `pb.inheritIO()`. However, if `EntryPoint.outPrintStream` is not null, it means that we have a custom redirection (on a custom instance `PrintStream`), and thus `pb.inheritIO()` does not the job, since `pb.inheritIO()` redirects on the caller's IO.

Why we do that? Because it allows the user to capture the IO of the process that runs the test, and keep in a Java Object String. Then, the user can play with this string, such as parsing and retrieve some info.

This feature is not a big deal since it is not used in DSpot, but since test-runner is a separated artifact and does not have the goal to be used only by DSpot, I think, if it possible, it would be great to keep this feature.

The worst case would be to open issue on the repository to report that this feature has been loss and we want it back.